audiobook_generator/tts_providers/openai_tts_provider.py

```diff
@@ -0,0 +1,91 @@
+import io
+import logging
+import math
+
+from openai import OpenAI
+
+from audiobook_generator.core.audio_tags import AudioTags
+from audiobook_generator.config.general_config import GeneralConfig
+from audiobook_generator.core.utils import split_text, set_audio_tags
+from audiobook_generator.tts_providers.base_tts_provider import BaseTTSProvider
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_supported_models():
+    return ["tts-1", "tts-1-hd"]
+
+
+def get_supported_voices():
+    return ["alloy", "echo", "fable", "onyx", "nova", "shimmer"]
+
+
+def get_supported_formats():
+    return ["mp3", "acc", "flac"]
+
+
+class OpenAITTSProvider(BaseTTSProvider):
+    def __init__(self, config: GeneralConfig):
+        super().__init__(config)
+        self.model = config.model_name or "tts-1"
+        self.voice = config.voice_name or "alloy"
+        self.format = config.output_format or "mp3"
+
+        self.price = 0.03 if self.model == "tts-1-hd" else 0.015
+        # per 1000 characters (0.03$ for HD model, 0.015$ for standard model)
+
+        self.client = OpenAI()  # User should set OPENAI_API_KEY environment variable
+
+    def __str__(self) -> str:
+        return super().__str__()
+
+    def text_to_speech(self, text: str, output_file: str, audio_tags: AudioTags):
+        max_chars = 4000  # should be less than 4096 for OpenAI
+
+        text_chunks = split_text(text, max_chars, self.config.language)
+
+        audio_segments = []
+
+        for i, chunk in enumerate(text_chunks, 1):
```

Comment:
Need to replace break_string/MAGIC_BREAK_STRING into blank space here. Or break symbols will be added into the AI voice.

I used to use three blank spaces here because I tested it sometimes would add more realistic break. But not sure about this because basically we can't control the break accurately in OpenAI TTS.

See https://github.com/p0n1/epub_to_audiobook/blob/f7fa674ef1e0f36660b87c0f5275d484498d0e9e/epub_to_audiobook.py#L236-L240.

Reply:
If you would inspect the code each TTS provider now has its own `MAGIC_BREAK_STRING`
for azure it remains: ` @BRK#`  but as your comment shows `OpenAI TTS doesn't support SSML ` it is set to 4 spaces by default

So this is not an issue any more and this piece fo code is redundant, that's why I deleted it.

Reply:
Oh, you're right. I missed that.